docs/tooling.md

```diff
@@ -32,11 +32,13 @@ We do have an artifact named `opensbi`, so the runner will check if it is alread
 ## Benchmark
 
 Different benchmark can be run with `just benchmark`.
-This command computes and prints some statistics about a run of Miralis.
+This command computes and prints some statistics about a run of Miralis with a given binary.
 
-The default behaviour is counting the number of retired instructions of function `run_vcpu` and `handle_trap` called in `main.rs` with the `default` firmware. You can specify the binary you want to benchmark with `just benchmark <binaries>`. For example, if you want to benchmark `linux` you can run `just benchmark linux`. 
+The binary should call Miralis to print the benchmark results at the end of its execution. Statistics are computed during the runtime in a stream maner.
 
-One can choose which benchmark to run in the `benchmark` section of the `config.toml` file. The field `enable` is overwritten to true when running this command.
+You can specify the binary you want to benchmark with `just benchmark <binaries>` (by default it is `ecall_benchmark`). For example, if you want to benchmark `linux` you can run `just benchmark linux`. 
 
-If you collect output of a run into a file, you can feed the file to the just `analyze-benchmark` command to get the statistics of the run.
+One can choose what to benchmark in the `benchmark` section of the `config.toml` file. The field `enable` is overwritten to true when running this command.
```

Comment:
It is not super clear what `enable` does just from reading the documentation.
You should probably invert the text, saying that benchmarks can be enables with `benchmark.enable`, that different values can be measures, and finally that `just benchmark` overrides `benchmark.enable` to `true`so it is equivalent to `just run` with a configuration that enables benchmark.

I have a related question: is there any difference now between using `just benchmark` and `just run` with a config that enables benchmarks?
I am pondering if we should stick with `just benchmark` or rather expose a configuration with benchmark enabled. Benchmarks only make sense on a board, but `just benchmark` is for QEMU, so I am just trying to evaluate if it is worth to keep it.

Reply:
I see, the command is still useful because it calls the runner with --benchmark arguments and a number of runs it has to do, in order to have an average of all runs. It is still only on qemu tho. If we don't need to benchmark on qemu then it is useless.